features/src/autogluon/features/generators/datetime.py

```diff
@@ -47,17 +46,27 @@ def get_default_infer_features_in_args() -> dict:
     def _compute_fillna_map(self, X: DataFrame):
         fillna_map = dict()
         for datetime_feature in self.features_in:
-            datetime_series = pd.to_datetime(X[datetime_feature], errors='coerce')
+            if type(X[datetime_feature].iloc[0]) == pd.Timestamp:
+                datetime_series = pd.to_datetime(X[datetime_feature], errors='coerce')
 
-            # Best guess is currently to fill by the mean.
-            fillna_datetime = datetime_series.mean()
-            fillna_map[datetime_feature] = fillna_datetime
+                # Best guess is currently to fill by the mean.
+                fillna_datetime = datetime_series.mean()
+                fillna_map[datetime_feature] = fillna_datetime
         return fillna_map
+    
+    def _get_country(self): # Obtain the country used to determine if it is a holiday or not.
+        if os.environ.get('LANG') == None:
```

Comment:
How about introducing `AG_COUNTRY`?  Also, you can use default return for getenv.

```
os.getenv('AG_COUNTRY', default='US')
```

LANG can have LANG=C and overriding LANG env is sometimes not preferred.
`os.environ.get('LANG').split("_")[1][0:2]` fails where `LANG=C`.

Reply:
@Innixma 

Is [_feature_generator_kwargs](https://github.com/awslabs/autogluon/blob/aab0dbef6de2132b1a74c92ecb93201e06f8f38f/tabular/src/autogluon/tabular/predictor/predictor.py#L652) right place for the toggle switch?  If it's not, let us know.
We consider adding constructor arguments to DatetimeFeatureGenerator based on [kwargs](https://github.com/awslabs/autogluon/blob/aab0dbef6de2132b1a74c92ecb93201e06f8f38f/features/src/autogluon/features/generators/auto_ml_pipeline.py#L75).

If `holiday_encoding(=jp)` is specified in `_feature_generator_kwargs`, try to import holiday package. Then, add holiday feature if holiday packages exists something like [try_import_mxboard](https://github.com/awslabs/autogluon/blob/fbdce82b4cdcaa4a1f2513c984e5aed91bbe1b51/core/src/autogluon/core/utils/try_import.py#L20
). Exclude holiday from setup.py.

Similary, enable adding sincos feature if `sincos_encoding(=True)` exists `_feature_generator_kwargs`.

BTW, @Kota28 is an intern student working at our company.